Encode strings using utf-8 when uploading

rvernica opened the pull request:
* Fix for #142 
* Measure string length after encoding
* Use `utf-8` encoding, override the default ASCII encoding in Python 2.7 (`utf-8` is the default in Python 3.x)
* Add test - upload non-ASCII data to SciDB

apoliakov commented:
Hey Rares, looks good in principle but I have some questions.

1. why does the thing say "all checks failed" in all but one commit? Is that an issue?

2. I checked this case and "mu" is actually "extended ascii" - it's ascii 230. https://theasciicode.com.ar/extended-ascii-code/lowercase-letter-mu-micro-sign-micron-ascii-code-230.html 
If unicode works, then it works. Just making sure. 

I'll wait for you to check on 1?

rvernica commented:
1. In the first commit, I fixed the bug and added a new test. The test kept failing for various reasons and I fixed it in subsequent commits. Finally, the test uncovered an issue with my fix for Python 2.7. That was fixed in the last commit and all the tests passed then. Already existing tests were passing, it was just the new test that killed the builds.
1. The ASCII codec for Python is for the base set not the extended set:
   ```python
   >>> 'µ'.encode('ascii')                                                      
   UnicodeEncodeError: 'ascii' codec can't encode character '\xb5' in position 0: ordinal not in range(128)
   ```
   Moreover, we always used `utf-8` for Python 3.x, as `encode()` was already in the code before this fix and the default is `utf-8` if no argument is provided. The underlying bug was that I miscalculated the length of the string when I was converting to binary.
